R/calculate_go_enrichment.R

```diff
@@ -55,24 +58,49 @@ go_enrichment <- function(...) {
+#' @param plot_style a character argument that specifies the plot style. Can be either "barplot" (default)
+#' or "heatmap". The "heatmap" plot is especially useful for the comparison of multiple groups. We recommend,
+#' however, that you use it only with `enrichment_type = "enriched"` or `enrichment_type = "deenriched`,
+#' because otherwise it is not possible to distinguish between enrichment and deenrichment in the plot.
+#' @param plot_title a character value that specifies the title of the plot. The default is "Gene ontology
+#' enrichment of significant proteins".
+#' @param barplot_fill_colour a vector that contains two colours that should be used as the fill colours for
+#' deenriched and enriched GO terms, respectively. If `enrichment_type = "enriched"` or `"deenriched`, please
+#' still provide two values in the vector, the colour not used for the plot can be `NA` in this case. E.g.
+#' `c(NA, "red")` for `enrichment_type = "enriched"`.
+#' @param heatmap_fill_colour a vector that contains colours that should be used to create the gradient in the
+#' heatmap plot. Default is `mako_colours`.
+#' @param heatmap_fill_colour_rev a logical value that specifies if the provided colours in `heatmap_fill_colour`
+#' should be reversed in order. Default is `TRUE`.
 #' @param label a logical argument indicating whether labels should be added to the plot.
 #' Default is TRUE.
+#' @param enrichment_type a character argument that is either "all", "enriched" or "deenriched". This
+#' determines if the enrichment analysis should be performed in order to check for both enrichemnt and
+#' deenrichemnt or only one of the two. This affects the statistics performed and therefore also the displayed
+#' plot.
+#' @param min_n_detected_proteins_in_process is a numeric argument that specifies the minimal number
+#' of detected proteins a GO term needs to have to be displayed in the plot. The default is 1, meaning
+#' no filtering of the plotted data is performed. This argument does not affect any computations or
+#' the returned data if `plot = FALSE`. This argument is useful in order to remove terms that were only
+#' detected in for example 1 protein. Even though these terms are sometimes significant, they are not
+#' really relevant.
 #' @param plot_cutoff a character value indicating if the plot should contain the top 10 most
 #' significant proteins (p-value or adjusted p-value), or if a significance cutoff should be used
 #' to determine the number of GO terms in the plot. This information should be provided with the
 #' type first followed by the threshold separated by a space. Example are
-#' \code{plot_cutoff = "adj_pval top10"}, \code{plot_cutoff = "pval 0.05"} or
-#' \code{plot_cutoff = "adj_pval 0.01"}. The threshold can be chosen freely. The default value is
-#' \code{"adj_pval top10"}.
+#' `plot_cutoff = "adj_pval top10"`, `plot_cutoff = "pval 0.05"` or
```

Comment:
I would add a small test in the function as well to check the input from the user. Something like this:

```
if (!grepl("^(pval|adj_pval) (top10|\\d+(\\.\\d+)?)$", plot_cutoff)) {
    stop("Invalid format for plot_cutoff. Please provide the type (pval or adj_pval) followed by 'top10' or a numeric threshold within the range (0, 1]. Valid formats are for instance: 'adj_pval top10', 'pval 0.05'")
  }```

Reply:
Thanks! I added this but also modified the plot_cutoff so that "top" can actually be used with any number.